### PR TITLE
Add simple query complexity cost calculations based on field costs

### PR DESCRIPTION
**Pull request by tkuhlman**

### CODE REVIEW

File: cost.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			return cost
+	}
+
+	for _, s := range field.SelectionSet.Selections {
+		if f, ok := s.(*ast.Field); ok {
+			fieldDef, ok := parent.Fields()[f.Name.Value]
+			if !ok {
+				continue
+			}
+			cost += astFieldCost(f, fieldDef)
```
Comment (enigmatic00) [inline cost.go:82]: for large queries, does this add enough overhead to warrant running the recursion in parallel?

**Review by zacaudette — APPROVED** [review]

**Review by enigmatic00 — APPROVED** [review]

File: cost.go (hunk 1 above)
Comment (udryan10) [reply cost.go:82]: 👍

Comment (tkuhlman) [reply cost.go:82]: I'm doing some basic performance tests with the content-api to get some numbers on the overhead added.

File: types.go [hunk 2]
```diff
@@ -7,8 +7,9 @@ import (
 // type Schema interface{}
 
 type Result struct {
-	Data   interface{}                `json:"data"`
-	Errors []gqlerrors.FormattedError `json:"errors,omitempty"`
+	Data            interface{}                `json:"data"`
+	Errors          []gqlerrors.FormattedError `json:"errors,omitempty"`
+	QueryComplexity int                        `json:"query_complexity,omitempty"`
```
Comment (enigmatic00) [inline types.go:12]: this should have been camelCase. sorry

Comment (tkuhlman) [reply types.go:12]: np, fixing

**Pull request merged**